synapse/models/inet.py

```diff
@@ -1442,7 +1442,12 @@ def getModelDefs(self):
                         'doc': 'A channel within a web service or instance such as slack or discord.'
                     }),
 
-                    ('inet:web:hashtag', ('str', {'lower': True, 'regex': r'^#\w[\w·]*(?<!·)$'}), {
+                    ('inet:web:hashtag', ('str', {'lower': True, 'regex': r'^#[^\p{P}\p{Z}]([^\p{Z}]*[^\p{P}\p{Z}])?$'}), {
+                        # regex explanation:
+                        # - starts with pound
+                        # - any non-punctuation/non-whitespace character
```

Comment:
Is non-punctuation starting character a twitter requirement? They don't allow 〜 as the first character?

Reply:
They do allow the wave dash as a first character. I can remove that requirement too. Although I realize now that (on twitter at least) emoji is not valid in hashtags. Not sure about other services though.

Reply:
Updated to pound followed by one or more non-whitespace characters. Will discuss as goback during scrum today.